src/payment/strategies/humm/humm-payment-strategy.ts

```diff
@@ -0,0 +1,59 @@
+import { FormPoster } from '@bigcommerce/form-poster';
+
+import { CheckoutStore,  InternalCheckoutSelectors } from '../../../checkout';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
+import { OrderFinalizationNotRequiredError } from '../../../order/errors';
+import { PaymentArgumentInvalidError } from '../../errors';
+import PaymentActionCreator from '../../payment-action-creator';
+import { PaymentRequestOptions } from '../../payment-request-options';
+import { ExternalPaymentStrategy } from '../external';
+
+export default class HummPaymentStrategy extends ExternalPaymentStrategy {
+
+    constructor(
+        _store: CheckoutStore,
+        _orderActionCreator: OrderActionCreator,
+        _paymentActionCreator: PaymentActionCreator,
+        _formPoster: FormPoster
+    ) {
+            super(
+            _store,
+            _orderActionCreator,
+            _paymentActionCreator,
+            _formPoster
+        );
+        }
+
+    async initialize(): Promise<InternalCheckoutSelectors> {
+        return this._store.getState();
+    }
+
+    async deinitialize(): Promise<InternalCheckoutSelectors> {
+        return this._store.getState();
+    }
+
+    async execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment, ...order } = payload;
+
+        if (!payment?.methodId) {
+            throw new PaymentArgumentInvalidError(['payment.methodId']);
+        }
+
+        await this._store.dispatch(this._orderActionCreator.submitOrder(order, options));
+        try {
+            return await this._store.dispatch(this._paymentActionCreator.submitPayment({methodId: payment.methodId}));
+        } catch (error) {
+            if (!error.body.provider_data || this._isAdditionalActionRequired(error)) {
+                return Promise.reject(error);
+            }
```

Comment:
🍹 One minor suggestion, not a blocker if you decide not to apply it:

If for some reason we don't receive the `provider_data`, the `additional action` will be thrown as a normal error and it won't have any sense to the shopper.

I'd recommend changing to this code:
```suggestion
            if (!this._isAdditionalActionRequired(error)) {
                return Promise.reject(error);
            }
            
            if (!error.body.provider_data) {
                throw new PaymentArgumentInvalidError(['payment. provider_data']);
            }
```